ext/opentelemetry-ext-pymongo/src/opentelemetry/ext/pymongo/__init__.py

```diff
@@ -82,7 +84,7 @@ def succeeded(self, event: monitoring.CommandSucceededEvent):
             span.set_attribute(
                 "db.mongo.duration_micros", event.duration_micros
             )
-            span.set_status(Status(StatusCanonicalCode.OK, event.reply))
+            span.set_status(OkStatus())
```

Comment:
Why do you discard the description here? This seems like a regression to me.

Reply:
Why updating pymongo integration at all this in the first place?, referenced issue you added is suggesting we do similar stuff as this code, having a default status of unknown when there is an error

Reply:
Ready, I added the description back. I understand how it can be valuable to have it to provide a better explanation of the status code, possibly to describe specific details of the code in its context.

Still, I am still changing the API because this PR changes the `Status` class from being instantiated with a particular code to having separated classes for every code itself. I believe this to be advantageous since when setting a status we go from this:

`set_status(Status(StatusCanonicalCode.OK, "description")))

to this:

`set_status(OkStatus("description"))`

Nevertheless, this may or may not be the desired approach for this project, please let me know in the comments.

Another important thing about this PR is the proposed dynamic generation of the statuses and their matching exceptions. Instead of having all them coded by hand, an `OrderedDict` is being used to define them and they are dynamically added to the module automatically. This saves code and saves us from having to keep the exceptions and the statuses synchronized by hand.

@lzchen points out [here](https://github.com/open-telemetry/opentelemetry-python/issues/292#issue-523580208) that in OpenCensus the status seems to be derived from the exception thrown. This implementation attempts to do that, by providing status-related exceptions that can be raised in the span code. For example, raising a `InvalidArgumentError("description")` exception in the span code automatically sets the span status to `InvalidArgumentStatus(description="description")`.

This may or may not be the desired approach for this project, please let me know in the comments.

Thanks!

Reply:
Irrespective of the merit of these changes, it is just best practice to submit independent changes in independent pull requests, at least when a controversial change is among them. Please split this PR up into:

* One that actually just automatically sets the status for ended spans on exception
* One that changes the Status API if you still want to do that

IMHO, this PR is mostly a discussion of the latter so it would make sense to make a new PR for (just) the exception Status thing and remove these changes from this PR.